tempto-core/src/main/java/com/teradata/tempto/internal/fulfillment/table/cassandra/CassandraTableManager.java

```diff
@@ -0,0 +1,228 @@
+
+@TableManager.Descriptor(tableDefinitionClass = RelationalTableDefinition.class, type = "CASSANDRA")
+@Singleton
+public class CassandraTableManager
+        implements TableManager<RelationalTableDefinition>
+{
+    private static final Logger LOGGER = getLogger(CassandraTableManager.class);
+
+    private static final int BATCH_SIZE = 10_000;
+
+    private final TableNameGenerator tableNameGenerator;
+    private final CassandraQueryExecutor queryExecutor;
+    private final String databaseName;
+    private final String defaultKeySpace;
+
+    @Inject
+    public CassandraTableManager(TableNameGenerator tableNameGenerator,
+            @Named("databaseName") String databaseName,
+            Configuration configuration)
+    {
+        this.tableNameGenerator = requireNonNull(tableNameGenerator, "tableNameGenerator is null");
+        this.queryExecutor = new CassandraQueryExecutor(configuration);
+        this.databaseName = requireNonNull(databaseName, "databaseName is null");
+        this.defaultKeySpace = configuration.getStringMandatory("databases.cassandra.default_schema");
+    }
+
+    @Override
+    public TableInstance<RelationalTableDefinition> createImmutable(RelationalTableDefinition tableDefinition, TableHandle tableHandle)
+    {
+        TableName tableName = createImmutableTableName(tableHandle);
+        checkArgument(tableName.getSchema().isPresent(), "Cassandra does not like you when you don't have keyspace defined");
+
+        if (!queryExecutor.tableExists(tableName.getSchema().get(), tableName.getInternalNameInDatabase())) {
+            executeQueryIgnoreError(tableDefinition.getCreateTableDDL(tableName.getNameInDatabase()));
+            RelationalDataSource dataSource = tableDefinition.getDataSource();
+
+            // TODO: check if data is up-to-date
+
+            insertData(tableName, dataSource);
+        }
+
+        return new CassandraTableInstance(tableName, tableDefinition);
+    }
+
+    private void insertData(TableName tableName, RelationalDataSource dataSource)
+    {
+        checkArgument(tableName.getSchema().isPresent(), "Cassandra does not like you when you don't have keyspace defined");
+
+        Iterator<List<Object>> dataRows = dataSource.getDataRows();
+        if (!dataRows.hasNext()) {
+            return;
+        }
+
+        List<String> columnNames = queryExecutor.getColumnNames(tableName.getSchema().get(), tableName.getInternalNameInDatabase());
+
+        try {
+            CassandraBatchLoader loader = new CassandraBatchLoader(queryExecutor.getSession(), tableName.getNameInDatabase(), columnNames);
+            for (List<List<Object>> batch : partitionBy(dataRows, BATCH_SIZE)) {
+                loader.load(batch);
+            }
+        }
+        catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static Iterable<List<List<Object>>> partitionBy(Iterator<List<Object>> dataRows, int partitionSize)
```

Comment:
It's only two places... And it has to be rewritten heavily when we'll introduce a proper modularization.